docs/contributing/coding-style.md

```diff
@@ -453,12 +453,20 @@ You can refer to [this section](#Struct-initialization) for more details of the
 
 We strongly recommend the following implementation to set the value using functional option.
 
+If an invalid value is set to the functional option, the `ErrInvalidOption` error defined in the [internal/errors/option.go](https://github.com/vdaas/vald/blob/master/internal/errors/option.go) should be returned.
+
+The name argument (the first argument) of the `ErrInvalidOption` error should be the same as the functional option name without the `With` prefix. 
+
+For example, the functional option name `WithVersion` should return the error with the argument `name` as `version`.
```

Comment:
LGTM, one point, It would be better to write about option load loop error handling using `errors.ErrOptionFailed`

Reply:
Thank you for your review.
I think so too.
But regarding error handling, I think it is better to add it to the proposal's PR because there is a possibility that it will change when the proposal is applied. 🤔

Reply:
I understand about it, but we should write current error handling use case first,  after proposal approved you should fix it

Reply:
It's not so difficult.

Reply:
thank you for your opinion.
I will add it to this PR soon